File: solutions/fsharp/raindrops/3/Raindrops.fs
module Raindrops

let convert (n: int) =
    let r =
        [| (3, "Pling"); (5, "Plang"); (7, "Plong") |]
        |> Seq.choose (fun (f, s) -> if n % f = 0 then Some s else None)

    if Seq.isEmpty r then n.ToString() else Seq.reduce (+) r

File: solutions/haskell/raindrops/3/src/Raindrops.hs
module Raindrops (convert) where

convert :: Int -> String
convert n =
  (printSounds . concatMap snd . filter (isDiv . fst)) sounds
  where
    sounds = [(3, "Pling"), (5, "Plang"), (7, "Plong")]
    isDiv f = n `mod` f == 0
    printSounds "" = show n
    printSounds res = res